### DISCUSSION TITLE
Slow and laggy interface

### DISCUSSION THREAD

**Discussion by thpauli** [bnpr/Malt#93] [Q&A] [answered]
Hi, 

I searched the discussions and could find and try similar issues, but not an answer to my case.

I'm on a good machine that runs cycles smoothly, but lags a lot when Malt has the render view activated on the viewport. 

I've tried downloading and replacing the python.exe file, as it was the solution suggested by @pragma37 to another user, but no success. My gpu is active, and I can spot a line that catched my eye on the malt log:

_Malt > Unable to load numpy_formathandler accelerator from OpenGL_accelerate_

At the same time, I could see that another user's log contains the same line, so I don't know if that's the problem.

Running Blender 2.93.1 portable on a nvme m.2 ssd drive, windows 10, intel i7 + rtx 3070 + rtx 3060

Here's the log
[malt 2021-06-27(21-30).log](https://github.com/bnpr/Malt/files/6722568/malt.2021-06-27.21-30.log)

Really eager to try Malt, the shader result I've seen are wonderful. 

Thanks in advance!

**Comment by pragma37** [comment, marked as answer]
Hi @thpauli !

Have you seen this thread #89?

There seems to be an issue with newer Nvidia cards where the GPU is not used at 100%.
Is that your case as well?

I have not been able to find the cause yet. As a temporal work-around I added an option to use a lower resolution scale, since it seems the problem is much worse at higher resolutions.

BTW,  Malt now ships with that custom Python version and is used automatically.
And that warning log shouldn't have any effect since we don't use numpy.

**Reply by thpauli** [reply]
Hi @pragma37 ! Thanks for your reply!

Glad to hear that the Numpy warning is valid, hah!

Yes, I've seen thread #89, but didn't think it would apply to me because I'm using 1920x1080 -- that'd be 1k, right? -- and not 4k resolution.

Then I downloaded today's release, uninstalled the previsou version and installed the new one. It's still very laggy, I can only get a smooth viewport when I scale it down to 0.4 resolution. 

GPU use peaks 100% when moving the camera around, tho.

**Reply by pragma37** [reply]
Really? That's extremely bad.
For reference, I have a gtx 950 and it runs ok at 1080.
1080 would be 2k, btw.

Could you test this?
https://github.com/bnpr/Malt/issues/89#issuecomment-870722997

My intuition is that this may be a memory bandwidth problem. Malt renders have to get downloaded from the GPU into the CPU, sent to Blender and then uploaded again to the GPU (thus the latency).
Maybe in your case is worse because your PCI bandwith have to be shared between GPUs?

**Reply by pragma37** [reply]
Could you check if this helps you?
https://github.com/bnpr/Malt/issues/89#issuecomment-881918060
